without-webhooks/client/ios-objc/app/CheckoutViewController.m

```diff
@@ -0,0 +1,208 @@
+    dispatch_async(dispatch_get_main_queue(), ^{
+        UIAlertController *alert = [UIAlertController alertControllerWithTitle:title message:message preferredStyle:UIAlertControllerStyleAlert];
+        if (restartDemo) {
+            [alert addAction:[UIAlertAction actionWithTitle:@"Restart demo" style:UIAlertActionStyleCancel handler:^(UIAlertAction *action) {
+                [self.cardTextField clear];
+                [self loadPage];
+            }]];
+        }
+        else {
+            [alert addAction:[UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleCancel handler:nil]];
+        }
+        [self presentViewController:alert animated:YES completion:nil];
+    });
+}
+
+- (void)loadPage {
+    // Load Stripe key from the server
+    NSURL *url = [NSURL URLWithString:[NSString stringWithFormat:@"%@stripe-key", BackendUrl]];
+    NSMutableURLRequest *request = [[NSURLRequest requestWithURL:url] mutableCopy];
+    [request setHTTPMethod:@"GET"];
+    [request setValue:@"application/json" forHTTPHeaderField:@"Content-Type"];
+    NSURLSessionTask *task = [[NSURLSession sharedSession] dataTaskWithRequest:request completionHandler:^(NSData *data, NSURLResponse *response, NSError *requestError) {
+        NSError *error = requestError;
+        NSDictionary *json = [NSJSONSerialization JSONObjectWithData:data options:0 error:&error];
+        NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *)response;
+        if (error != nil || httpResponse.statusCode != 200 || json[@"publicKey"] == nil) {
+            [self displayAlertWithTitle:@"Error loading page" message:error.localizedDescription ?: @"" restartDemo:NO];
+        }
+        else {
+            NSLog(@"Loaded Stripe key");
+            NSString *stripePublicKey = json[@"publicKey"];
+            [Stripe setDefaultPublishableKey:stripePublicKey];
+        }
+    }];
+    [task resume];
+}
+
+- (void)pay {
+    // Collect card details on the client
+    STPPaymentMethodCardParams *cardParams = self.cardTextField.cardParams;
+    STPPaymentMethodParams *paymentMethodParams = [STPPaymentMethodParams paramsWithCard:cardParams billingDetails:nil metadata:nil];
+    [[STPAPIClient sharedClient] createPaymentMethodWithParams:paymentMethodParams completion:^(STPPaymentMethod *paymentMethod, NSError *createError) {
+        // Create PaymentMethod failed
+        if (createError != nil) {
+            [self displayAlertWithTitle:@"Payment failed" message:createError.localizedDescription ?: @"" restartDemo:NO];
+        }
+        else if (paymentMethod != nil) {
+            // Create a PaymentIntent on the server with a PaymentMethod
+            NSLog(@"Created PaymentMethod");
+            [self confirmPaymentMethod:paymentMethod.stripeId orPaymentIntent:nil];
+        }
+    }];
+}
+
+- (void)confirmPaymentMethod:(NSString *)paymentMethodId orPaymentIntent:(NSString *)paymentIntentId {
```

Comment:
nit: I find this name confusing since `confirmPaymentMethod` isn't a thing, you can't confirm a PaymentMethod.  I think I'd create separate `createAndConfirmPaymentIntent` and `reconfirmPaymentIntent` methods here.